device/peer.go

```diff
@@ -17,17 +17,16 @@ import (
 
 type Peer struct {
 	isRunning         atomic.Bool
-	sync.RWMutex      // Mostly protects endpoint, but is generally taken whenever we modify peer
 	keypairs          Keypairs
 	handshake         Handshake
 	device            *Device
-	endpoint          conn.Endpoint
+	endpoint          atomic.Value   // conn.Endpoint or nil
```

Comment:
this comment kinda implies it can go back and forth, which atomic.Value doesn't permit.

Maybe `// of conn.Endpoint (but initially nil)`